ui/desktop/app/routes/scopes/scope/projects/sessions/session.js

```diff
@@ -17,9 +17,25 @@ export default class ScopesScopeProjectsSessionsSessionRoute extends Route {
    * Load a session
    * @param {object} params
    * @param {string} params.session_id
-   * @return {TargetModel}
+   * @return {{session: SessionModel, target: TargetModel}}
    */
-  model({ session_id }) {
-    return this.store.findRecord('session', session_id, { reload: true });
+  async model({ session_id }) {
+    let target = null;
+    const session = await this.store.findRecord('session', session_id, {
+      reload: true,
+    });
+
+    try {
+      if (session.target_id) {
```

Comment:
What happens when we have no session? 
Do we want to check for that here? `session?.target_id`

Reply:
This is actually part of the work that I believe will be changing with the "Connect" refactor. I am going to start on that next since it is a large chunk of this work. Good idea to add a check though that if we don't find the session for some reason we redirect the user.